image_eval/evaluators.py

```diff
@@ -1,100 +1,135 @@
+        torch_imgs = [transforms.ToTensor()(img).to(self.device) for img in generated_images]
         self.evaluator.update(torch_imgs, prompts)
-        return {"clip_score": self.evaluator.compute()}
+        return {"clip_score": self.evaluator.compute().item()}
+
 
+class StyleSimilarityEvaluator(BaseEvaluator):
+    TYPE = EvaluatorType.FIDELITY
+    HIGHER_IS_BETTER = True
 
-class StyleSimilarityEvaluator(BaseWithReferenceEvaluator):
     def __init__(self, device: str):
         super().__init__(device)
-        self.encoders = [encoder_cls(device) for encoder_cls in ALL_ENCODER_CLASSES]
 
-    def evaluate(self, generated_images: list[Image.Image], real_images: list[Image.Image]):
+    def evaluate(self,
+                 generated_images: list[Image.Image],
+                 real_images: list[Image.Image],
+                 **ignored_kwargs) -> Dict[str, float]:
         """Returns the average cosine similarity between the generated images and the center of the cluster defined by real images."""
         cos = torch.nn.CosineSimilarity(dim=1, eps=1e-6)
         results = {}
 
-        for encoder in self.encoders:
+        for encoder_cls in ALL_ENCODER_CLASSES:
+            encoder = encoder_cls(self.device)
             generated_embeddings = encoder.encode(generated_images)
             generated_center = torch.mean(generated_embeddings, axis=0, keepdim=True)
             real_embeddings = encoder.encode(real_images)
             real_center = torch.mean(real_embeddings, axis=0, keepdim=True)
-            results[f"style_similarity_{encoder.id}"] = cos(generated_center, real_center)
+            results[f"style_similarity_{encoder.id}"] = cos(generated_center, real_center).item()
         return results
 
 
-class InceptionScoreEvaluator(BaseReferenceFreeEvaluator):
+class InceptionScoreEvaluator(BaseEvaluator):
+    TYPE = EvaluatorType.IMAGE_QUALITY
+    HIGHER_IS_BETTER = True
+
     def __init__(self, device: str):
         super().__init__(device)
         self.evaluator = InceptionScore().to(self.device)
 
-    def evaluate(self, images: list[Image.Image], ignored_prompts: list[str]):
-        torch_imgs = torch.stack([transforms.ToTensor()(img).to(torch.uint8).to(self.device) for img in images])
+    def evaluate(self, generated_images: list[Image.Image], **ignored_kwargs) -> Dict[str, float]:
+        torch_imgs = torch.stack([transforms.ToTensor()(img).to(torch.uint8).to(self.device)
+                                  for img in generated_images])
         self.evaluator.update(torch_imgs)
-        return {"inception_score": self.evaluator.compute()}
+        mean, stddev = self.evaluator.compute()
+        return {"inception_score_mean": mean.item(),
+                "inception_score_stddev": stddev.item()}
+
+    def is_useful(self, generated_images: list[Image.Image], **ignored_kwargs) -> bool:
+        # InceptionScore calculates a marginal distribution over the objects identified in the
```

Comment:
It would be good to include a link or citation for this observation (paper, blog post, etc)